torchinductor/utils.py

```diff
@@ -165,3 +165,50 @@ def wrapper(self):
         return getattr(self, key)
 
     return wrapper
+
+
+def sympy_str(expr: sympy.Expr):
+    """
+    Normal sympy str is very slow, this is a lot faster.  The result are
+    somewhat worse, as it doesn't do as much simplification.  So don't
+    use this for final codegen.
+    """
+    if isinstance(expr, sympy.Symbol):
+        return expr.name
+    if isinstance(expr, sympy.Add):
+        return " + ".join(map(sympy_str, expr.args))
+    if isinstance(expr, sympy.Mul):
+        return " * ".join(map(sympy_str, expr.args))
+
+    from .ir import CleanDiv
+    from .ir import IndexingDiv
+    from .ir import ModularIndexing
+
+    if isinstance(expr, (ModularIndexing, CleanDiv, IndexingDiv)):
+        return f"{expr.func.__name__}({', '.join(map(sympy_str, expr.args))})"
+    return str(expr)
+
+
+def sympy_subs(expr: sympy.Expr, replacements: Dict[Any, Any]):
+    """
+    xreplace is faster than subs, but is way more picky
+    """
+
+    def swap_key(key):
+        if isinstance(key, str):
+            return sympy.Symbol(key)
+        assert isinstance(key, sympy.Symbol), key
+        return key
+
+    def swap_value(val):
+        if isinstance(val, int):
+            return sympy.Integer(val)
```

Comment:
I'm curious why this is needed. I checked and it looks like `xreplace` automatically converts `int` to `Integer`.

Reply:
You are correct, it may not be needed.